ivy/array/array.py

```diff
@@ -701,16 +701,7 @@ def __irshift__(self, other):
 
     @_native_wrapper
     def __deepcopy__(self, memodict={}):
-        try:
-            return to_ivy(self._data.__deepcopy__(memodict))
-        except AttributeError:
-            # ToDo: try and find more elegant solution to jax inability to
-            #  deepcopy device arrays
-            if ivy.current_backend_str() == "jax":
-                np_array = copy.deepcopy(self._data)
-                jax_array = ivy.array(np_array)
-                return to_ivy(jax_array)
-            return to_ivy(copy.deepcopy(self._data))
+        return to_ivy(self._data.__deepcopy__(memodict))
```

Comment:
keep the old `try` `except` stuff here, this error might only arise when using Device arrays on the GPU for example. Are you sure the error never occurs for JAX?

Reply:
I thought there was a fix as I went through Jax Repository and found out that they solved the issue here [Deepcopy issue](https://github.com/google/jax/issues/2632), [Pull Request](https://github.com/google/jax/pull/10659). But if it is for GPU case I can revert the changes

Reply:
I also tried it on google Colab running on GPU and seemed to work

Reply:
As `bfloat16` is not still supported we remain with this exception handling.